syntax = "proto3";

message Settings {
    enum node_type {
        TYPE_NONE = 0;
        UNDEF = 1;
        ANCHOR = 2;
        SYNC_NODE = 3;
        TAG = 4;
    };
    enum rtls_mode {
        MODE_NONE = 0;
        OFF = 1;
        TWR_INITIATOR = 2;
        TWR_RESPONDER = 3;
        TDOA_ANCHOR = 4;
        TDOA_SYNC = 5;
        TDOA_TAG = 6;
    };

    node_type   NodeType    = 1;
    int32      DeviceID    = 2;
    rtls_mode   RTLSMode    = 3;
    int32      NodeID      = 4;  // ID in UNE

    bytes  NetworkMAC       = 5;
    uint32 DeviceIp         = 6;  // Network IP
    uint32 SubnetMask       = 7;  // Network Sunbet Mask
    uint32 GatewayIp        = 8;  // Network Gateway IP
    uint32 ServerIp         = 9;  // Network Server IP
    uint32 ConnectionPort   = 10;  // Network Connection Port

    float PositionX = 11;   // Anchor X
    float PositionY = 12;   // Anchor Y
    float PositionZ = 13;   // Anchor Z

    int32 DWRxAntDelay = 14;       // Antenna receive delay for DW1000 modules, -1 = zero delay
    int32 DWTxAntDelay = 15;       // Antenna transmit delay for DW1000 modules, -1 = zero delay

    repeated uint32 TwrConnectedAnchors = 16;  // List of anchors connected with tag, Max count: 10
    uint32          TwrPollPeriod       = 17;  // Period of tag poll messages in ms
    int32           TwrPollDelay        = 18;  // Delay between polls for diffenet anchors -1 = no delay

    int32          TdoaSyncPeriod      = 19;  // Syncronisation period of sync node blinks
    int32          TdoaPollPeriod      = 20;  // Period of tag poll messages in ms
    int32          TdoaSyncSenderID    = 21;  // Id of prev sync node
    int32          TdoaSyncReceiverID  = 22;  // Id of next sync node
}


